### PR TITLE
add nb_conda

### PR DESCRIPTION
**Pull request by bollwyvl**
This is the in-Jupyter-notebook conda env manager:
- https://github.com/Anaconda-Server/nb_conda

### CODE REVIEW

**Comment by bollwyvl** [comment]
Updated this, as per:

https://github.com/conda-forge/conda-forge.github.io/issues/173#issuecomment-224806476

Still has the browser-based tests in it, as these do represent the minimum test stance for achieving user functionality (they don't care if it's REST or websockets or carrier pigeons).

Happy to make any additional changes to get this merged!

File: recipes/nb_conda/yum_requirements.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+fontconfig
```
Comment (ocefpaf) [inline recipes/nb_conda/yum_requirements.txt]: I am having a *deja vu* here (maybe you already explained this to me sorry for asking again :grimacing:) but why `conda-forge`'s `fontconfig` does not work for here?

Comment (jakirkham) [inline recipes/nb_conda/yum_requirements.txt]: No worries. I believe it was `phantomjs` in the other cases. Though I don't see it in the requirements list.

Do we actually need this, @bollwyvl?

Comment (bollwyvl) [inline recipes/nb_conda/yum_requirements.txt]: Yes, it is a test-time dependency for phantomjs.

At some point we need to circle back around for a phantomjs recipe:
https://github.com/conda-forge/staged-recipes/pull/692

...but in the meantime, this gets the job done.

**Comment by bollwyvl** [comment]
ea05355 was pushed because of: https://github.com/conda-forge/nbpresent-feedstock/issues/3

**Comment by bollwyvl** [comment]
Tests will almost certainly fail on windows until this finishes:
https://ci.appveyor.com/project/conda-forge/nb-conda-kernels-feedstock/build/1.0.4

**Comment by bollwyvl** [comment]
Oy. Don't know what to do about this. To get nb_conda_kernels to [build with conda 4.1.1](https://github.com/conda-forge/nb_conda_kernels-feedstock/pull/7) on windows, I ended up having to change the appveyor script to:
```
"%CMD_IN_ENV% conda build -c defaults -c conda-forge recipe --quiet"
```
to ensure that I pulled the correct combination of [vs2015-runtime and python](https://github.com/conda/conda/issues/2727).

I don't know how to do that with conda-build-all here.

Thoughts? @jakirkham @ocefpaf

**Comment by bollwyvl** [comment]
Ok, turns out by doing the `defaults` config after updating conda, everything worked out. It is mostly building off defaults, but of course only on windows. I'm told that per-package channel pinning is [coming](https://github.com/conda/conda/issues/2727#issuecomment-226665461), which might make some stuff better...

**Comment by bollwyvl** [comment]
Any feedback here? Have we learned more from the conda 4.1.x issues? I see 4.1.2 is out.

File: appveyor.yml [hunk 2]
```diff
@@ -72,6 +72,8 @@ install:
     - cmd: conda config --add channels conda-forge
     - cmd: conda config --set show_channel_urls true
     - cmd: conda update --yes --quiet conda
+    - cmd: conda config --remove channels defaults
+    - cmd: conda config --add channels defaults
```
Comment (ocefpaf) [inline appveyor.yml]: Is this still needed or was it fixed in `conda-config`?

**Comment by ocefpaf** [comment]
> Any feedback here? Have we learned more from the conda 4.1.x issues? I see 4.1.2 is out.

@bollwyvl don't known the first thing about packaging jupyter extensions, but this looks good to me.

**Comment by jakirkham** [comment]
`conda config` bug should already be fixed in the recent release. It has been demonstrated to work in a few cases.

**Comment by bollwyvl** [comment]
cool, i'll try reverting my hacks to appveyor.yml

**Comment by bollwyvl** [comment]
The new vs2014 didn't solve the post-link problem:
https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.3506/job/d4p3oafqd8626dvx#L491

Investigating locally...

**Comment by bollwyvl** [comment]
The new vs2014 didn't solve the issue: looks like we need zeromq as well... or need to reconsider this whole approach.

**Comment by pelson** [comment]
Anything I can do to help move this forward @bollwyvl?

**Comment by msarahan** [comment]
Reopening to test with conda-build 1.21.6, which adds a PATH change to try to make the post-link scripts here work.

**Comment by bollwyvl** [comment]
I am assuming this is still blocked by:
https://github.com/conda-forge/nb_conda_kernels-feedstock/pull/8

Which is in turn blocked by:
https://github.com/conda-forge/pyzmq-feedstock/pull/8

**Comment by bollwyvl** [comment]
Blar, download issues on circle. Will let the others run to completion, see what's what!

**Comment by bollwyvl** [comment]
Travis will probably fail until this is resolved:
https://github.com/conda-forge/nb_conda_kernels-feedstock/issues/10

**Comment by bollwyvl** [comment]
Hoooo da lally! Never thought I'd see the day. Would love some review on this to make sure I'm up to speed on all the developments recently. Thanks for everyone's help!

**Comment by bollwyvl** [comment]
@ocefpaf @jakirkham @pelson anything else I can do to push this forward?

**Comment by ocefpaf** [comment]
I think it is past time we get this into a `feedstock`. Any remaining issue/update can be done there. Thanks @bollwyvl and sorry for the delay to notice this here. Too much noise in `conda-forge` lately.

**Comment by bollwyvl** [comment]
Wooo! Thanks so much!

>  Too much noise in conda-forge lately.

Don't 🔥, man: you guys are doing a fantastic job!

**Pull request merged**